src/devices/Display/Ht16k33.cs

```diff
@@ -0,0 +1,227 @@
+    /// </summary>
+    /// <remarks>
+    /// Datasheet: https://www.adafruit.com/datasheets/ht16K33v110.pdf
+    /// Sources:
+    /// https://github.com/adafruit/Adafruit_LED_Backpack/blob/master/Adafruit_LEDBackpack.cpp
+    /// https://github.com/sobek1985/Adafruit_LEDBackpack/blob/master/Adafruit_LEDBackpack/AlphaNumericFourCharacters.cs
+    /// </remarks>
+    public abstract partial class Ht16k33 : IDisposable
+    {
+        #region Constants
+        /// <summary>
+        /// HT16K33 default I2C address
+        /// </summary>
+        public const int DefaultI2cAddress = 0x70;
+
+        /// <summary>
+        /// Maximum level of brightness
+        /// </summary>
+        public const byte MaxBrightness = 15;
+        #endregion
+
+        #region Protected members
+        /// <summary>
+        /// I2C device interface
+        /// </summary>
+        protected I2cDevice I2cDevice;
+
+        /// <summary>
+        /// AutoFlush mode
+        /// </summary>
+        protected bool autoFlush = true;
+        #endregion
+
+        #region Private members
+        /// <summary>
+        /// Display on/off state
+        /// </summary>
+        private bool _displayOn = true;
+
+        /// <summary>
+        /// Initial brightness level
+        /// </summary>
+        private byte _brightness;
+
+        /// <summary>
+        /// Initial blinking rate
+        /// </summary>
+        private BlinkRate _blinkRate;
+
+        /// <summary>
+        /// Send turn on oscillator command to device
+        /// </summary>
+        private void TurnOnOscillator() =>
+            I2cDevice.WriteByte((byte)Command.TurnOnOscillator);
+
+        /// <summary>
+        /// Send set brightness command to device
+        /// </summary>
+        private void SetBrightness() =>
+            I2cDevice.WriteByte((byte)((byte)Command.SetBrightness | _brightness & (byte)Command.BrightnessMask));
+
+        /// <summary>
+        /// Send Blink/DisplayOn command to device
+        /// </summary>
+        private void SetBlinkDisplayOn() =>
+            I2cDevice.WriteByte((byte)((byte)Command.Blink | (byte)Command.BlinkDisplayOn | (byte)_blinkRate << 1));
+        #endregion
+
+        #region Public members
+        /// <summary>
+        /// Initialize HT16K33 device
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="System.Device.I2c.I2cDevice"/> to create with.</param>
+        public Ht16k33(I2cDevice i2cDevice)
+        {
+            // Initialize I2C device
+            I2cDevice = i2cDevice;
+
+            TurnOnOscillator();
+            SetBrightness();
+            SetBlinkDisplayOn();
+        }
+
+        /// <summary>
+        /// Gets or sets whether buffering is enabled
+        /// </summary>
+        /// <remarks>When set to true the display buffer is only flushed to the device when explicitly calling the <see cref="Flush"/> method</remarks>
+        public bool BufferingEnabled
```

Comment:
this can be just `{ get; set; }`

Reply:
Overoptimized a little bit, but I guess I should leave that to the compiler :)